pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -89,7 +91,14 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env model.Environmen
 
 			// create default cluster
 			clusterName := model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port)
+			upstreamServiceAccounts := env.ServiceAccounts.GetIstioServiceAccounts(service.Hostname, []string{port.Name})
 			defaultCluster := buildDefaultCluster(env, clusterName, convertResolution(service.Resolution), hosts)
+
+			// set TLSSettings if configmap global settings specifies MUTUAL_TLS.
+			if env.Mesh.AuthPolicy == meshconfig.MeshConfig_MUTUAL_TLS {
+				applyUpstreamTLSSettings(defaultCluster, buildIstioMutualTls(upstreamServiceAccounts))
```

Comment:
Bug: this cannot be applied for external services. Add a check for service.MeshExternal and skip setting up this istio mutual for default

Reply:
I also don’t know how this will work for headless services (ie ones using original dst cluster). You have to enable the auth in headless service test and see. If it fails add a check for original dst cluster here as well, just like the plugin code

Reply:
I checked external service now. Sorry for the lack of familiarity, what's the headless services/original dst cluster? is there some field to check, similar as `service.MeshExternal`?